src/components/Trigger/Tests/basic.test.jsx

```diff
@@ -0,0 +1,883 @@
+            trigger(container, '.target');
+
+            const clickPopupDomNode =
+                document.querySelector('.click-trigger').parentElement;
+            const hoverPopupDomNode =
+                document.querySelector('.hover-trigger').parentElement;
+            expect(clickPopupDomNode).toBeTruthy();
+            expect(hoverPopupDomNode).toBeTruthy();
+
+            trigger(container, '.target', 'mouseLeave');
+            expect(
+                hoverPopupDomNode.className.includes('-hidden')
+            ).toBeTruthy();
+            expect(clickPopupDomNode.className.includes('-hidden')).toBeFalsy();
+
+            fireEvent.click(container.querySelector('.target'));
+            act(() => jest.runAllTimers());
+            expect(
+                hoverPopupDomNode.className.includes('-hidden')
+            ).toBeTruthy();
+            expect(
+                clickPopupDomNode.className.includes('-hidden')
+            ).toBeTruthy();
+        });
+    });
+
+    // Placement & Align can not test in jsdom. This should test in `dom-align`
+
+    describe('forceRender', () => {
+        it("doesn't render at first time when forceRender=false", () => {
+            render(
+                <Trigger popup={<span>Hello!</span>}>
+                    <span>Hey!</span>
+                </Trigger>
+            );
+
+            expect(document.querySelector('.trigger-popup')).toBeFalsy();
+        });
+
+        it('render at first time when forceRender=true', () => {
+            const triggerRef = React.createRef();
+            class Test extends React.Component {
+                render() {
+                    return (
+                        <Trigger
+                            ref={triggerRef}
+                            forceRender
+                            popup={<span>Hello!</span>}
+                        >
+                            <span>Hey!</span>
+                        </Trigger>
+                    );
+                }
+            }
+            render(<Test />);
+            expect(triggerRef.current.getPopupDomNode()).toBeTruthy();
+        });
+    });
+
+    describe('destroyPopupOnHide', () => {
+        it('defaults to false', () => {
+            const triggerRef = createRef();
+            const { container } = render(
+                <Trigger
+                    ref={triggerRef}
+                    action={['click']}
+                    popupAlign={placementAlignMap.topRight}
+                    popup={<strong>trigger</strong>}
+                >
+                    <div className="target">click</div>
+                </Trigger>
+            );
+
+            trigger(container, '.target');
+            expect(triggerRef.current.getPopupDomNode()).toBeTruthy();
+
+            trigger(container, '.target');
+            expect(triggerRef.current.getPopupDomNode()).toBeTruthy();
+        });
+
+        // TODO: Re-enable test when destroyPopupOnHide bug is fixed
```

Comment:
what is this bug you speak of. is it tracked somewhere and can we reference that so some future contributor has more context?